### PR TITLE
chore: make followRedirect option accept functions

### PR DESCRIPTION
**Pull request by VanTigranyan**
#### Checklist

- [x] I have read the documentation.
- [x] I have included a pull request description of my changes.
- [x] I have included some tests.
- [x] If it's a new feature, I have included documentation updates in both the README and the types.

### CODE REVIEW

**Comment by VanTigranyan** [comment]
With these changes, it is possible to pass the `followRedirect` option as a function that gets a PlainResponse right before the redirect is done and returns true or false. By that, it is possible to decide whether to redirect dynamically. The option supports both plain boolean and function type, allowing users to decide for themselves which type they prefer.

File: source/core/options.ts [hunk 1]
```diff
@@ -1763,12 +1763,12 @@ export default class Options {
 
 	@default true
 	*/
-	get followRedirect(): boolean {
+	get followRedirect(): boolean | ((response: PlainResponse) => boolean) {
```
Comment (sindresorhus) [inline source/core/options.ts:1766]: The function needs to be documented (with text) here and in the readme.

File: .gitignore [hunk 2]
```diff
@@ -4,3 +4,4 @@ coverage
 .nyc_output
 dist
 *.0x
+.idea
```
Comment (sindresorhus) [inline .gitignore:7]: This doens't belong here.

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "got",
-	"version": "13.0.0",
+	"version": "13.1.0",
```
Comment (sindresorhus) [inline package.json:3]: This should be done by the maintainers.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: source/core/options.ts (hunk 1 above)
Comment (VanTigranyan) [reply source/core/options.ts:1766]: By documented you mean I should add a note how to use it?

File: .gitignore (hunk 2 above)
Comment (VanTigranyan) [reply .gitignore:7]: Will remove

File: package.json (hunk 3 above)
Comment (VanTigranyan) [reply package.json:3]: Got it. Would be nice to have a CONTRIBUTING.md with mentions what and where we should do. I'll remove this.

**Comment by VanTigranyan** [comment]
@sindresorhus Made the fixes you asked for. Thanks for the quick review.

**Comment by VanTigranyan** [comment]
@sindresorhus BTW: I would be happy to contribute by adding a semantic release setup that would increase the version of the lib based on commit messages and would publish automatically.

File: package.json (hunk 3 above)
Comment (sindresorhus) [reply package.json]: I don't see the need to document basic contribution things in every project. Generally, you should try to touch the least amount of code and config.

**Comment by sindresorhus** [comment]
> I would be happy to contribute by adding a semantic release setup that would increase the version of the lib based on commit messages and would publish automatically.

No thanks. I prefer the human touch to relasing and changelogs.

**Review by sindresorhus — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (VanTigranyan) [reply package.json]: Got it. Thanks.

**Pull request merged**